src/layers/models/layersManager.ts

```diff
@@ -163,7 +187,7 @@ export class LayersManager {
 
         // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
       } else if (jobType === JobAction.UPDATE) {
-        const record = await this.catalog.findRecord(productId, undefined, productType);
+        const record = await asyncCallInSpan(async () => this.catalog.findRecord(productId, undefined, productType), this.tracer, 'findRecord');
```

Comment:
Don't we need to add "handleSpanOnError"  on "if (!existsInMapProxy) .... throw new BadRequestError(message);" several lines ahead?

Reply:
and the same here. this code is under the same try-catch block